org.eclipse.xtext.maven.plugin/src/main/java/org/eclipse/xtext/maven/AbstractXtextGeneratorMojo.java

```diff
@@ -137,6 +137,14 @@ public String getEncoding() {
 	@Parameter(defaultValue = "false")
 	private Boolean incrementalXtextBuild = Boolean.FALSE;
 
+	/**
+	 * Will use the ResourceStorage to write the semantic model,
+	 * {@link IResourceDescription} and optional the node model also to a binary
+	 * file. For details see {@link StorageAwareResource} and {@link ResourceStorageFacade}.
+	 */
```

Comment:
can you please add a `@since 2.35`

Reply:
Maybe the Javadoc needs some adjustments if I understand the intended meaning correctly. Shouldn't it be something like

```
Whether to use the ResourceStorage (is it a link to Java type?) to write the semantics model, the resource descriptions {@link IResourceDescription}, and optionally...
```
```

Reply:
Nit: I see that we use `Boolean` instead of `boolean` for some properties in this class. Should be fine to use the primitive `boolean` here, as it seems.

Reply:
@szarnekow should this be addressed in this PR?

Reply:
@LorenzoBettini 
Should i change my Boolean to the primitiv type?

Reply:
@florianpirchner That's what I asked @szarnekow . If you don't mind, you could do that in this PR. Maybe also the other `Boolean`s.

Reply:
@LorenzoBettini sure. i will do